### PR TITLE
build: adds a .envrc.local file for local development for dotenv

### PR DESCRIPTION
**Pull request by faiq**
**What problem does this PR solve?**:

allows developers to have `.envrc.local` files to contain secret information like `NUTANIX_USER` and `NUTANIX_PASSWORD` 
based on this https://github.com/direnv/direnv/issues/544#issuecomment-598654584

**Which issue(s) this PR fixes**:
Fixes #

**How Has This Been Tested?**:

**Special notes for your reviewer**:

### CODE REVIEW

File: .envrc [hunk 1]
```diff
@@ -7,5 +7,7 @@ export DEVBOX_NO_ENVRC_UPDATE=1
 
 eval "$(devbox generate direnv --print-envrc --env-file .dev-envrc)"
 
+if [[ -f .envrc.local ]]; then dotenv ".envrc.local"; fi
```
Comment (jimmidyson) [inline .envrc:10]:
```suggestion
dotenv_if_exists '.envrc.local'
```

Comment (faiq) [reply .envrc:10]: wowza

**Review by supershal — APPROVED** [review]
LGTM

**Review by jimmidyson — APPROVED** [review]

File: .envrc (hunk 1 above)
[the remark this answers is not in the document]
Comment (faiq) [reply .envrc]: i'd really prefer to not touch that file as much as possible. i think this solves  an issue for right now.

**Comment by supershal** [comment]
we can have a helper makefile target that can generate `envrc.local` from caren e2e config. https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/blob/main/test/e2e/config/caren.yaml#L137

**Pull request merged**